src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,47 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String allDate = getAllDataBetweenDate(data, dateFrom, dateTo);
+        StringBuilder builder = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo + System.lineSeparator());
+        String[] employees = allDate.split(";");
+        for (String name : names) {
+            int salary = 0;
+            for (String dataEmployee : employees) {
+                String[] employee = dataEmployee.split(" ");
+                if (employee[0].equals(name)) {
+                    int time = Integer.parseInt(employee[1]);
+                    int money = Integer.parseInt(employee[2]);
```

Comment:
indexes are magic number, should be constant

```suggestion
                if (employee[0].equals(name)) {
                    int time = Integer.parseInt(employee[1]);
                    int money = Integer.parseInt(employee[2]); 
```

Reply:
But they are change every iteration.